src/utils/testing.rs

```diff
@@ -224,439 +170,386 @@ pub async fn empty_processed_alerts_queue(
+    object_id: Option<String>, // Use String for all, convert as needed
     ra: Option<f64>,
     dec: Option<f64>,
 }
 
-#[async_trait::async_trait]
-impl AlertRandomizerTrait for ZtfAlertRandomizer {
-    type ObjectId = String;
-
-    fn default() -> Self {
-        let payload = fs::read("tests/data/alerts/ztf/2695378462115010012.avro").unwrap();
-        let reader = Reader::new(&payload[..]).unwrap();
-        let schema = reader.writer_schema().clone();
-        Self {
-            payload: Some(payload),
-            schema: Some(schema),
-            candid: Some(Self::randomize_i64()),
-            object_id: Some(Self::randomize_object_id()),
-            ra: Some(Self::randomize_ra()),
-            dec: Some(Self::randomize_dec()),
-        }
-    }
-
-    fn new() -> Self {
+impl AlertRandomizer {
+    pub fn new(survey: Survey) -> Self {
         Self {
+            survey,
             payload: None,
             schema: None,
+            schema_registry: None,
             candid: None,
             object_id: None,
             ra: None,
             dec: None,
         }
     }
 
-    fn path(mut self, path: &str) -> Self {
-        let payload = fs::read(path).unwrap();
-        let reader = Reader::new(&payload[..]).unwrap();
-        let schema = reader.writer_schema().clone();
-        self.payload = Some(payload);
-        self.schema = Some(schema);
-        self
-    }
-
-    fn objectid(mut self, object_id: impl Into<Self::ObjectId>) -> Self {
-        self.object_id = Some(object_id.into());
-        self
-    }
-
-    fn candid(mut self, candid: i64) -> Self {
-        self.candid = Some(candid);
-        self
-    }
-
-    fn ra(mut self, ra: f64) -> Self {
-        match Self::validate_ra(ra) {
-            true => self.ra = Some(ra),
-            false => panic!("RA must be between 0 and 360"),
-        }
-        self
-    }
-    fn dec(mut self, dec: f64) -> Self {
-        match Self::validate_dec(dec) {
-            true => self.dec = Some(dec),
-            false => panic!("Dec must be between -90 and 90"),
-        }
-        self
-    }
-
-    fn rand_object_id(mut self) -> Self {
-        self.object_id = Some(Self::randomize_object_id());
-        self
-    }
-    fn rand_candid(mut self) -> Self {
-        self.candid = Some(Self::randomize_i64());
-        self
-    }
-    fn rand_ra(mut self) -> Self {
-        self.ra = Some(Self::randomize_ra());
-        self
-    }
-    fn rand_dec(mut self) -> Self {
-        self.dec = Some(Self::randomize_dec());
-        self
-    }
-
-    async fn get(self) -> (i64, Self::ObjectId, f64, f64, Vec<u8>) {
-        let mut candid = self.candid;
-        let mut object_id = self.object_id;
-        let mut ra = self.ra;
-        let mut dec = self.dec;
-        let (payload, schema) = match (self.payload, self.schema) {
-            (Some(payload), Some(schema)) => (payload, schema),
-            _ => {
+    pub fn default(survey: Survey) -> Self {
```

Comment:
FYI the `default` method is normally associated with the `Default` trait, so it's a little unexpected to define a constructor with this same name when it was nothing to do with `Default`. What we have here could be called `new_from _survey` or possibly just `new`. Feel free to leave this as-is, I'm just calling out that I've been spotting this around the codebase.

Reply:
So I'm unsure what to do here. There is a `new` method already, that just sets the survey and nothing else. On the other hand default does not just set the survey but all the other parameters, with randomized new values and such. In theory, if these are not set they get autogenerated in the `get` method anyways, so perhaps `default` is just useless here. The issue is I can't implement the default trait to call a default method because that takes parameters as input, when the default trait does not want any.

Reply:
maybe I should just have a `randomize_all` option that does exactly what default does, that someonce calls after calling new() ?

Reply:
Ok I'll just rename `default` by `new_randomized`

Reply:
Yep, that's the right thing to do.